src/ddscxx/tests/QosProvider.cpp

```diff
@@ -165,6 +165,33 @@ class QosProvider : public ::testing::Test
       enum duration_unit reader_data_lifecycle_disposed_unit;
     } sysdef_qos_conf_t;
 
+constexpr static const unsigned char base64_etable[64] = {
+  'A','B','C','D','E','F','G','H','I','J','K','L','M','N','O','P','Q','R','S','T','U',
+  'V','W','X','Y','Z','a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p',
+  'q','r','s','t','u','v','w','x','y','z','0','1','2','3','4','5','6','7','8','9','+','/'
+};
+
+static uint32_t b64_encode (const unsigned char *text, const uint32_t sz, unsigned char **buff)
+{
+  uint32_t act_len = (sz * 4U/3U); 
+  uint32_t buff_len = (act_len % 4U) != 0u? ((act_len / 4U + 1U)*4U): act_len;
+  *buff =  static_cast<unsigned char *>(ddsrt_malloc(buff_len));
+  (void) memset (*buff, '=', buff_len);
+
+  for (size_t i = 0, j = 0; i < buff_len && j < sz; i+=4, j+=3)
+  {
+    unsigned char chunk[4] = {0x00, 0x00, 0x00, 0x00};
+    chunk[0] = base64_etable[(text[j] >> 0x02U)];
+    chunk[2] = base64_etable[((text[j+1] & 0x0FU) << 0x02U) | (text[j+2] & 0xC0U) >> 0x06U];
```

Comment:
This (`text[j+1]` and `text[j+2]`) can overrun the input buffer, and it does: https://dev.azure.com/eclipse-cyclonedds/cyclonedds-cxx/_build/results?buildId=6049&view=logs&j=adc611be-d2d6-58a9-dbb6-278cbdb63baf&t=45723d4c-ae75-5df9-8dab-ef8105b34d21

Reply:
fixed 8e3d1d7